OptimizelySDKCore/OptimizelySDKCore/OPTLYProjectConfig.m

```diff
@@ -182,6 +183,22 @@ - (OPTLYAttribute *)getAttributeForKey:(NSString *)attributeKey {
     return attribute;
 }
 
+- (nullable NSString *)getAttributeIdForKey:(nonnull NSString *)attributeKey {
+    OPTLYAttribute *attribute = self.attributeKeyToAttributeMap[attributeKey];
+    BOOL hasReservedPrefix = [attributeKey hasPrefix:kReservedAttributePrefix];
+    NSString *attributeId;
+    if (attribute) {
```

Comment:
One condition should be covered, if attributeKey is not reserved and not in datafile. Looks missing to me.